server/src/main/java/org/elasticsearch/index/mapper/SourceValueFetcher.java

```diff
@@ -60,7 +60,7 @@ public SourceValueFetcher(String fieldName, QueryShardContext context, Object nu
         for (String path : sourcePaths) {
             Object sourceValue = lookup.extractValue(path, nullValue);
             if (sourceValue == null) {
-                return List.of();
+                continue;
```

Comment:
This was a bug - if the first of multiple source paths didn't have any values, we returned early instead of continuing to check further paths.